pandas/index.pyx

```diff
@@ -371,6 +482,16 @@ cdef class IndexEngine:
 
         return result[0:count], missing[0:count_missing]
 
+    def get_indexer_non_unique_orderable(self, ndarray targets,
+                                         int64_t[:] idx0,
+                                         int64_t[:] idx1):
```

Comment:
Notice that I have removed the code regarding lists. Most of the code consists in two functions: `_count` and `_map`. They implement two very specific algorithms that I couldn't find in the above-mentioned file.